lib/ImageHelper.php

```diff
@@ -179,6 +179,17 @@ protected static function add_actions() {
 				}
 			}
 		} );
+		add_filter('wp_generate_attachment_metadata', function( $metadata, $post_id ) {
+			$post = get_post($post_id);
+			$image_types = array('image/jpeg', 'image/png', 'image/gif', 'image/jpg');
+			if ( in_array($post->post_mime_type, $image_types) ) {
+				$attachment = new Image($post_id);
+				if ( $attachment->file_loc ) {
+					ImageHelper::delete_generated_files($attachment->file_loc);
+				}
+			}
+			return $metadata;
+		} );
```

Comment:
The filter uses practically the same code as the `delete_attachment` action above. Why not extract the logic inside the action and the filter as a new method of the `ImageHelper` class?